frb_codegen/src/config.rs

```diff
@@ -78,30 +79,109 @@ pub struct Opts {
-            .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("rust_crate_dir")))
-    }));
-    let manifest_path = {
-        let mut path = std::path::PathBuf::from_str(&rust_crate_dir).unwrap();
-        path.push("Cargo.toml");
-        path_to_string(path).unwrap()
-    };
-    let rust_output_path = canon_path(&raw.rust_output.unwrap_or_else(|| {
-        fallback_rust_output_path(&rust_input_path)
-            .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("rust_output")))
-    }));
-    let class_name = raw.class_name.unwrap_or_else(|| {
-        fallback_class_name(&*rust_crate_dir)
-            .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("class_name")))
+pub fn parse(raw: RawOpts) -> Vec<Opts> {
+    // rust input path(s)
+    let mut rust_input_paths = raw.rust_input.clone();
+    rust_input_paths.retain(|each_path| !each_path.trim().is_empty());
+    let rust_input_paths = rust_input_paths
+        .iter()
+        .map(|each_path| canon_path(each_path))
+        .collect::<Vec<_>>();
+    assert!(
+        rust_input_paths.len() >= 1,
+        "rust input(s) should have at least 1 path"
+    );
+
+    // dart output path(s)
+    let mut dart_output_paths = raw.dart_output.clone();
+    dart_output_paths.retain(|each_path| !each_path.trim().is_empty());
+    let dart_output_paths = dart_output_paths
+        .iter()
+        .map(|each_path| canon_path(each_path))
+        .collect::<Vec<_>>();
+    assert!(
+        dart_output_paths.len() == rust_input_paths.len(),
+        "dart output path(s) should have the same number of path(s) as rust input(s)"
+    );
+
+    // rust crate dir(s)
+    let rust_crate_dirs = raw.rust_crate_dir.unwrap_or_else(|| {
+        rust_input_paths
+            .iter()
+            .map(|each_rust_input_path| {
+                fallback_rust_crate_dir(each_rust_input_path)
+                    .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("rust_crate_dir")))
+            })
+            .collect::<Vec<_>>()
     });
-    let c_output_path = raw
+    let rust_crate_dirs = rust_crate_dirs
+        .iter()
+        .map(|each_path| canon_path(each_path))
+        .collect::<Vec<_>>();
+    assert!(
+        rust_crate_dirs.len() == rust_input_paths.len(),
+        "rust crate dir(s) should have the same number of path(s) as rust input(s)"
+    );
+
+    // manifest path(s)
+    let manifest_paths = rust_crate_dirs
+        .iter()
+        .map(|each| {
+            let mut path = std::path::PathBuf::from_str(each).unwrap();
+            path.push("Cargo.toml");
+            path_to_string(path).unwrap()
+        })
+        .collect::<Vec<_>>();
+    assert!(
+        manifest_paths.len() == rust_input_paths.len(),
+        "manifest path(s) should have the same number of path(s) as rust input(s)"
+    );
+
+    // rust output path(s)
+    let rust_output_paths = match raw.rust_output {
+        Some(rust_outputs) => rust_outputs,
```

Comment:
Is `Option.unwrap_or_else` simpler?

Reply:
Fixed in 9e38c2995f796d3e5c35b6bbaa631d62645325dc.